table/metadata.go

```diff
@@ -399,3 +400,32 @@ func (m *MetadataV2) UnmarshalJSON(b []byte) error {
 	m.preValidate()
 	return m.validate()
 }
+
+func NewMetadataV2(schema *iceberg.Schema, partitionSpec iceberg.PartitionSpec, sortOrder SortOrder, location string, tableUUID uuid.UUID, properties iceberg.Properties) (*MetadataV2, error) {
```

Comment:
Should we use a builder pattern like we do for manifests? Instead of a single func with a ton of params?

Reply:
@zeroshade Yeah i thought about that, the only "optional" thing is properties, some have "empty values" like sort order and partition spec so it may work OK.

I will try out a builder and post an update.